src/test/java/seedu/address/model/session/RecurringSessionTest.java

```diff
@@ -1,18 +1,65 @@
 package seedu.address.model.session;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.SessionBuilder.DEFAULT_TIME;
+import static seedu.address.testutil.TypicalStudents.CARL;
 
 import org.junit.jupiter.api.Test;
 
-class RecurringSessionTest {
-    // TODO: Add RecurringSessionTest and other parse tests.
+class RecurringSessionTest extends SessionTest {
+    static final Interval WEEKLY = new Interval("7");
+    static final SessionDate SESSION_DATE = new SessionDate("2021-01-01", "10:00");
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new RecurringSession(null,
+                null, null, null, null, null));
+    }
+
+    @Test
+    public void isValidEnd() {
+        SessionDate consistent = new SessionDate("2021-01-15", "10:00");
+        SessionDate inconsistentDate = new SessionDate("2021-01-14", "10:00");
+        SessionDate inconsistentTime = new SessionDate("2021-01-15", "12:00");
+
+
+        assertTrue(RecurringSession.isValidEnd(SESSION_DATE, consistent, WEEKLY));
+        assertFalse(RecurringSession.isValidEnd(SESSION_DATE, inconsistentDate, WEEKLY));
+        assertFalse(RecurringSession.isValidEnd(SESSION_DATE, inconsistentTime, WEEKLY));
+    }
+
+    @Test
+    public void isConsistentDatesAndInterval() {
+        SessionDate consistent1 = new SessionDate("2021-01-15", "12:00");
+        SessionDate same = new SessionDate("2021-01-01", "10:00");
+        SessionDate inconsistentDate = new SessionDate("2021-01-14", "10:00");
+        Interval inconsistentInterval = new Interval("3");
+        Interval nextDay = new Interval("1");
+
+
+        assertTrue(RecurringSession.isConsistentDatesAndInterval(SESSION_DATE, consistent1, WEEKLY));
+        assertFalse(RecurringSession.isConsistentDatesAndInterval(SESSION_DATE, same, nextDay));
+        assertFalse(RecurringSession.isConsistentDatesAndInterval(SESSION_DATE, inconsistentDate, WEEKLY));
+        assertFalse(RecurringSession.isConsistentDatesAndInterval(SESSION_DATE, consistent1, inconsistentInterval));
+    }
 
     @Test
     void getInterval() {
```

Comment:
Hi, maybe using something like "getIntervalTest" might make it more explicate that its a method for testing.